src/nools/lib.js

```diff
@@ -1,45 +1,53 @@
-if (c.contact && c.contact.type === 'person') {
-  for(idx1=0; idx1<targets.length; ++idx1) {
-    target = targets[idx1];
-    switch(target.appliesToType) {
-      case 'person':
-        emitPersonBasedTargetFor(c, target);
-        break;
-      case 'report':
-        for(idx2=0; idx2<c.reports.length; ++idx2) {
-          r = c.reports[idx2];
-          emitReportBasedTargetFor(c, r, target);
-        }
-        break;
-      default:
-        throw new Error('unrecognised target type: ' + target.type);
+var place = ['health_center', 'district_hospital', 'clinic'];
+
+for(idx1=0; idx1<targets.length; ++idx1) {
+  target = targets[idx1];
+  if(target.appliesTo === 'contacts') {
+    if(c.contact && c.contact.type === 'person' &&
+      target.appliesToType.includes('person')) {
+      emitPersonBasedTargetFor(c, target);
     }
-  }
-  for(idx1=0; idx1<tasks.length; ++idx1) {
-    // TODO currently we assume all tasks are report-based
+    if(c.contact && place.includes(c.contact.type) &&
+      target.appliesToType.includes(c.contact.type)) {
+      emitPlaceBasedTargetFor(c, target);
+    }
```

Comment:
Seems as though the person and place can be combined to avoid duplicate code. Also, we can make it more generic by not defining specific place types, which may be useful if/when we have more place types in the near future.

eg
```js
if(target.appliesTo === 'contacts') {
  if(c.contact && target.appliesToType.includes(c.contact.type)) {
    emitContactBasedTargetFor(c, target);
  }
} else if(target.appliesTo === 'reports') {
```

Reply:
We also need to check that we can use `includes` here too.